src/Patches.cs

```diff
@@ -169,51 +196,78 @@ private static void RecordData(string line, string informationType, bool append)
 
 	internal class LootTableHelper
 	{
-		internal static string GetLootTableName(Container container)
+		internal static string? GetLootTableName(Container? container)
 		{
-			if (container is null) return null;
+			if (container == null) return null;
 
-			if (container.IsLocked() && container.m_LockedLootTablePrefab != null) return container.m_LockedLootTablePrefab.name;
+			if (container.IsLocked() && container.m_LockedLootTableData != null) return container.m_LockedLootTableData.name;
 
-			if (container.m_LootTablePrefab != null) return container.m_LootTablePrefab.name;
+			if (container.m_LootTableData != null) return container.m_LootTableData.name;
 
 			return null;
 		}
 	}
 
-	[HarmonyPatch(typeof(PlayerManager), "GetInteractiveObjectDisplayText")]
-	internal class PlayerManager_GetInteractiveObjectDisplayText
+	[HarmonyPatch(typeof(PlayerManager), nameof(PlayerManager.GetInteractiveObjectUnderCrosshairs), new Type[] { typeof(float) })]
+	internal class GetInteractiveObjectUnderCrosshairs
 	{
-		public static void Postfix(PlayerManager __instance, ref string __result)
+		public static void Postfix(ref GameObject? __result)
 		{
-			if (__instance?.m_InteractiveObjectUnderCrosshair?.name is null) return;
-			if (__result is null || string.IsNullOrWhiteSpace(GameManager.m_ActiveScene)) return;
+			if (__result == null || string.IsNullOrWhiteSpace(GameManager.m_ActiveScene))
+			{
+				KeyTracker.ResetLooatingAt();
+				return;
+			}
+
+			if (Settings.options.useDeleteFunction && CustomInput.GetKeyDown(Settings.options.deleteKey))
+			{
+				UnityEngine.Object.Destroy(__result);
+				KeyTracker.ResetLooatingAt();
+				__result = null;
+				return;
+			}
+
+			if (KeyTracker.lookingAt == __result || KeyTracker.currentMode == GrabberMode.None)
+			{
+				return;
+			}
+
+			string outputMsg = string.Empty;
```

Comment:
This assignment can be removed if a default case is added to the following switch statement.
```cs
default:
    return;
```

Reply:
Doesn't work as the switch conditions require the variable to be defined previously,
`The name 'outputMsg' does not exist in the current context`
And setting all conditions to `string ` doesn't work either.
`A local variable or function named 'outputMsg' is already defined in this scope`

Reply:
```cs
			string outputMsg;
			switch (KeyTracker.currentMode)
			{
				case GrabberMode.Name:
					outputMsg = "\nname = " + __result.name;
					break;
				case GrabberMode.Position:
					outputMsg = "\nposition = " + FormatHelper.FormatVector(__result.transform.position);
					break;
				case GrabberMode.Rotation:
					outputMsg = "\nrotation = " + FormatHelper.FormatVector(__result.transform.rotation.eulerAngles);
					break;
				case GrabberMode.Scene:
					outputMsg = "\nscene = " + GameManager.m_ActiveScene;
					break;
				case GrabberMode.LootTable:
					Container container = __result.GetComponentInChildren<Container>();
					if (container != null)
					{
						outputMsg = "\nloottable = " + LootTableHelper.GetLootTableName(container);
					}
					else
					{
						return;
					}
					break;
				default:
					return;
			}
```

Reply:
I misunderstood this as remove the whole line, which obviusly wouldn't work.
Sorted.